modules/api/test/LorisApiInstrumentsTest.php

```diff
@@ -0,0 +1,933 @@
+            gettype($InstrArray['Meta']['DDE']),
+            'boolean'
+        );
+        $this->assertSame(
+            gettype($InstrArray['Meta']['Instrument']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($InstrArray[$this->instrumentTest]['CommentID']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($InstrArray[$this->instrumentTest]['Testdate']),
+            'string'
+        );
+        $this->assertSame(
+            gettype(
+                $InstrArray[$this->instrumentTest]['Data_entry_completion_status']
+            ),
+            'string'
+        );
+
+        $this->assertArrayHasKey(
+            'Candidate',
+            $InstrArray['Meta']
+        );
+        $this->assertArrayHasKey(
+            'Visit',
+            $InstrArray['Meta']
+        );
+        $this->assertArrayHasKey(
+            'DDE',
+            $InstrArray['Meta']
+        );
+        $this->assertArrayHasKey(
+            'Instrument',
+            $InstrArray['Meta']
+        );
+        $this->assertArrayHasKey(
+            'CommentID',
+            $InstrArray[$this->instrumentTest]
+        );
+        $this->assertArrayHasKey(
+            'UserID',
+            $InstrArray[$this->instrumentTest]
+        );
+        $this->assertArrayHasKey(
+            'Examiner',
+            $InstrArray[$this->instrumentTest]
+        );
+        $this->assertArrayHasKey(
+            'Testdate',
+            $InstrArray[$this->instrumentTest]
+        );
+        $this->assertArrayHasKey(
+            'Data_entry_completion_status',
+            $InstrArray[$this->instrumentTest]
+        );
+    }
+
+    /**
+     * Tests the HTTP PATCH request for the
+     * endpoint /projects/{project}/instruments{instrument}
+     *
+     * @return void
+     */
+    public function testPatchCandidatesCandidVisitInstrumentsInstrument(): void
+    {
+        $candid     = '300004';
+        $visit      = 'V1';
+        $instrument = 'aosi';
+        $json       = [
+            'Meta'      => [
+                'CandID'     => $candid,
+                'Visit'      => $visit,
+                'DDE'        => false,
+                'Instrument' => $instrument
+            ],
+            $instrument => [
+                'UserID' => "2"
+            ]
+
+        ];
+        $response = $this->client->request(
+            'PATCH',
+            "candidates/$candid/$visit/instruments/$instrument",
+            [
+                'http_errors' => false,
+                'headers'     => $this->headers,
+                'json'        => $json
+            ]
+        );
+        if ($response->getStatusCode() === 404) {
+            $this->markTestIncomplete(
+                "Endpoint not found: PATCH" .
+                "candidates/$candid/$visit/instruments/$instrument"
+            );
+        }
+        if ($response->getStatusCode() === 405) {
+            $this->markTestSkipped(
```

Comment:
It should be be skipped. If a 405 is returned then the test must fail.

Reply:
not failing, I will remove